odcockpit/src/plugins/environmentviewer/EnvironmentViewerGLWidget.cpp

```diff
@@ -442,6 +449,49 @@ namespace cockpit {
                         glEnd();
                     }
     */
+                    //Visualize compact point cloud, where points are sorted by increasing azimuth (increment the azimuth after every 16 points) and 
+                    //vertical angle (from -15 to 15 with increment 2 for each 16 points).
+                    //A compact point cloud contains: (1) the starting azimuth, (2) the ending azimuth, (3) number of points per azimuth, 
+                    //and (4) a string with the distance values of all points in the container
+                    if (m_CPCreceived) {
+                        Lock lockCPC(m_cpcMutex);
+                        float startAzimuth = m_cpc.getStartAzimuth();
+                        float endAzimuth = m_cpc.getEndAzimuth();
+                        uint8_t entriesPerAzimuth = m_cpc.getEntriesPerAzimuth();
+                        string distances = m_cpc.getDistances();
+                        uint32_t numberOfPoints = distances.size()/2;
+                        uint32_t numberOfAzimuths = numberOfPoints/entriesPerAzimuth;
+                        float azimuthIncrement = (endAzimuth-startAzimuth)/numberOfAzimuths;//Calculate the azimuth increment
+                        stringstream sstr(distances);
+                        
+                        m_root->render(m_renderingConfiguration); 
```

Comment:
m_root->render(...) is called above already in one of the two other if-branches. Could you make this if-block either a subsequent else-if block and keep m_root->render or remove it here to avoid double rendering?